static/js/composebox_typeahead.js

```diff
@@ -315,9 +317,9 @@ exports.content_typeahead_selected = function (item) {
     if (this.completing === 'emoji') {
         // leading and trailing spaces are required for emoji, except if it begins a message.
         if (beginning.lastIndexOf(":") === 0 || beginning.charAt(beginning.lastIndexOf(":") - 1) === " ") {
-            beginning = beginning.replace(/:\S+$/, "") + ":" + item.emoji_name + ": ";
+            beginning = (beginning.substring(0, beginning.length - this.token.length-1)+ ":" + item.emoji_name + ": ");
         } else {
-            beginning = beginning.replace(/:\S+$/, "") + " :" + item.emoji_name + ": ";
+            beginning = (beginning.substring(0, beginning.length - this.token.length-1) + " :" + item.emoji_name + ": ");
```

Comment:
I am using the `substring` method to return a substring of `beginning` till the starting of the emoji name. 
This is a better way than using a regex as the expression gets complicated when using spaces. 


And sorry about the spacing near subtraction, it missed my eye somehow.